lso/applications/objective_functions/squares/squares_area.py

```diff
@@ -0,0 +1,46 @@
+import numpy as np
+from typing import Optional
+
+from lso.utils import area_utils
+from lso.data import data as lso_data
+from lso.objective_function import objective_function as lso_objective_function
+
+
+class SquaresArea(lso_objective_function.ObjectiveFunction):
```

Comment:
In general - I think that this class follows the following pattern of a bad Objective (not Procedural) design:
- you name a general object (`SquaresArea`),
- you place a huge switch between different methods and implement these methods inside class body and function call.

This is naturally not an objective but procedural programming. In this case it might not work well because you conflate a lot of different objects with a high probability of extending them in the future what will end up in either exponential boom of amount of code needed (and needless code repetition [Btw. I do not claim that all code repetition is bad, sometimes it is really good, but IMHO - this is not the case]).

What I see here is an ontology of the following classes:

class RectangleObjectiveFunction
class RectangleArea(RectangleObjectiveFunction)
class RectangleLikeliness(RectangleObjectiveFunction)
class CompositeObjectiveFunction(ObjectiveFunction) <- this class accepts a list of pairs (ObjectiveFunction and Weight) and returns weighted objective function.

Reply:
Move to applications/square_data

Reply:
JacardPenalizedShapeObjective

Reply:
Changed the name to JacardPenalizedShapeObjective

Reply:
Removed the switch